src/feature/hs/hs_cache.c

```diff
@@ -659,14 +683,18 @@ cache_store_as_client(hs_cache_client_descriptor_t *client_desc)
    * client authorization. */
   cache_entry = lookup_v3_desc_as_client(client_desc->key.pubkey);
   if (cache_entry != NULL) {
-    /* Signalling an undecrypted descriptor. We'll always replace the one we
-     * have with the new one just fetched. */
-    if (cache_entry->desc == NULL) {
+    /* Without a decrypted descriptor, we'll always replace the one we have
+     * with the new one we just fetched because we can't inspect the revision
+     * counter in the plaintext data. */
+    if (!entry_has_decrypted_descriptor(client_desc)) {
       remove_v3_desc_as_client(cache_entry);
       cache_client_desc_free(cache_entry);
       goto store;
     }
 
+    /* From this point on, we know that the decrypted descriptor is in the
```

Comment:
Ye it's kinda weird that now the check of ->desc is not explicit so these kind of comments are helpful.... Anyhow nothing to change here, just mentioning this weirdness.